perl/streaming_pipe_in_and_out.pl

```diff
@@ -0,0 +1,17 @@
+# Script reads text from a named pipe and writes it another named pipe, capitalized
+use strict;
+use warnings;
+
+my ($pipe_in, $pipe_out) = @ARGV;
+
+open my $output, '>', $pipe_out or die "Cannot open output pipe: $!";
+open my $input, '<', $pipe_in or die "Cannot open input pipe: $!";
+
+select((select($output), $| = 1)[0]);
```

Comment:
```suggestion
$output->autoflush(1);
```

This accomplishes the same thing without the shenanigans that `select` performs.